### PR TITLE
Some fixes

### PR DESCRIPTION
**Pull request by killercup**

### CODE REVIEW

**Comment by soc** [comment]
@killercup Thanks ... I should have documented the state of testing on various platforms. Welp.

Basically:
- Linux: Compiles and runs correctly.
- Windows: Compiles.
- macOS: Not tested.

File: src/mac.rs [hunk 1]
```diff
@@ -58,10 +59,10 @@ impl ProjectDirectories {
     }
 
     pub fn from_project_name(project_name: &str) -> ProjectDirectories {
-        from_unprocessed_string(name)
+        ProjectDirectories::from_unprocessed_string(project_name.to_string())
```
Comment (soc) [inline src/mac.rs]: Is the `to_string()` necessary here?

Looking at e. g. https://github.com/soc/directories-rs/blob/master/src/win.rs#L72 the same code works without it.

Comment (killercup) [reply src/mac.rs]: You're calling `from_unprocessed_string` which expects a String. I have not looked into code any further :)

Comment (killercup) [reply src/mac.rs]: Oh, on Linux, the first thing `from_unprocessed_string` does is `let project_name = String::from(value);`, so it's also allocating a String there.

You could probably use `Cow<str>` but I don't think it's a huge performance loss.

**Comment by soc** [comment]
@killercup Thanks, I changed a few things around, checked that it compiles and pushed your changes!

**Pull request closed without merging**